app/assets/javascripts/initializers/initializeReadingListIcons.js

```diff
@@ -83,17 +86,22 @@ function renderButtonState(button, json) {
 }
 
 function renderNewSidebarCount(button, json) {
-  var newCount;
-  var count = document.getElementById('reading-list-count').dataset.count;
-  count = parseInt(count, 10);
-  if (json.result === 'create') {
-    newCount = count + 1;
-  } else if (count !== 0) {
-    newCount = count - 1;
+  const readingListContainers = document.querySelectorAll(
+    '.reading-list-count',
+  );
+  if (readingListContainers) {
+    readingListContainers.forEach(function (e) {
+      let count = parseInt(e.dataset.count, 10);
```

Comment:
```suggestion
      const count = parseInt(e.dataset.count, 10);
```
(just a suggestion / not blocking) - I don't think we ever reassign this